dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/support/ProviderConsumerRegTable.java

```diff
@@ -92,4 +92,29 @@ public static void registerConsumer(Invoker invoker, URL registryUrl, URL consum
         return invokers;
     }
 
+    public static boolean isRegistered(String serviceUniqueName) {
+        Set<ProviderInvokerWrapper> providerInvokerWrapperSet = ProviderConsumerRegTable.getProviderInvoker(serviceUniqueName);
+        for (ProviderInvokerWrapper providerInvokerWrapper : providerInvokerWrapperSet) {
+            if (providerInvokerWrapper.isReg()) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public static int getConsumerAddressNum(String serviceUniqueName) {
+        int count = 0;
+        Set<ConsumerInvokerWrapper> providerInvokerWrapperSet = ProviderConsumerRegTable.getConsumerInvoker(serviceUniqueName);
```

Comment:
can be replace by 

return providerInvokerWrapperSet.stream()
                .map(piw->piw.getRegistryDirectory().getUrlInvokerMap())
                .filter(Objects::nonNull)
                .mapToInt(Map::size).sum();

Reply:
done

Reply:
done

Reply:
Thanks.